Add ongamepadconnected + ongamepaddisconnected

marcoscaceres opened the pull request:
#### 57f8b6aa692023a6ee9e78c9d6fe73e9c571e75b
<pre>
Add ongamepadconnected + ongamepaddisconnected
<a href="https://bugs.webkit.org/show_bug.cgi?id=223860">https://bugs.webkit.org/show_bug.cgi?id=223860</a>

Reviewed by Brady Eidson.

* LayoutTests/gamepad/gamepad-event-handlers-expected.txt: Added.
* LayoutTests/gamepad/gamepad-event-handlers.html: Added.
* LayoutTests/gamepad/gamepad-polling-access-expected.txt:
* LayoutTests/gamepad/gamepad-polling-access.html:
* LayoutTests/gamepad/gamepad-visibility-1.html:
* LayoutTests/platform/gtk/TestExpectations
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/gamepad/WindowEventHandlers+Gamepad.idl: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/HTMLAttributeNames.in:

Canonical link: <a href="https://commits.webkit.org/252331@main">https://commits.webkit.org/252331@main</a>
</pre>

dcrousso commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -17887,6 +17891,9 @@
 		E7FBE7C124EB2FAB00A18E62 /* StereoPannerNode.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = StereoPannerNode.h; sourceTree = "<group>"; };
 		E7FBE7C424EC36F300A18E62 /* StereoPanner.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = StereoPanner.h; sourceTree = "<group>"; };
 		E7FBE7C524EC36F300A18E62 /* StereoPanner.cpp */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.cpp.cpp; path = StereoPanner.cpp; sourceTree = "<group>"; };
+		E886AA61285AC5A5008FF9F7 /* WindowEventHandlers+Gamepad.idl */ = {isa = PBXFileReference; lastKnownFileType = text; name = "WindowEventHandlers+Gamepad.idl"; path = "Modules/gamepad/WindowEventHandlers+Gamepad.idl"; sourceTree = "<group>"; };
```
I think this should just have a `path` 
```
E886AA61285AC5A5008FF9F7 /* WindowEventHandlers+Gamepad.idl */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text; path = "WindowEventHandlers+Gamepad.idl"; sourceTree = "<group>"; };
```

dcrousso commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -19117,6 +19124,7 @@
 		0867D691FE84028FC02AAC07 /* WebCore */ = {
 			isa = PBXGroup;
 			children = (
+				E886AA61285AC5A5008FF9F7 /* WindowEventHandlers+Gamepad.idl */,
```
i think this should be in the `518F4FE9194CA4B60081BAAE /* gamepad */` group below

dcrousso commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -5913,6 +5915,8 @@
 			dstSubfolderSpec = 16;
 			files = (
+				E8FC921C285BFD5B004904B2 /* JSWindowEventHandlers+Gamepad.cpp in Product Dependencies */,
+				E8FC921B285BFD5B004904B2 /* JSWindowEventHandlers+Gamepad.h in Product Dependencies */,
```
i dont think this is needed/wanted

dcrousso commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -5531,6 +5531,8 @@
 		E7E0357224D4E196008DFEFB /* AnalyserOptions.h in Headers */ = {isa = PBXBuildFile; fileRef = E7E0357124D4E191008DFEFB /* AnalyserOptions.h */; };
 		E7FBE7C224EB2FB000A18E62 /* StereoPannerNode.h in Headers */ = {isa = PBXBuildFile; fileRef = E7FBE7C124EB2FAB00A18E62 /* StereoPannerNode.h */; };
 		E7FBE7C324EB2FB600A18E62 /* StereoPannerOptions.h in Headers */ = {isa = PBXBuildFile; fileRef = E7FBE7C024EB2FAB00A18E62 /* StereoPannerOptions.h */; };
+		E8FC921B285BFD5B004904B2 /* JSWindowEventHandlers+Gamepad.h in Product Dependencies */ = {isa = PBXBuildFile; fileRef = E8FC9219285BFD5A004904B2 /* JSWindowEventHandlers+Gamepad.h */; };
```
I think this needs to be added to `93F198A608245E59001E9ABC /* Headers */`.  I think you can do that by selecting this file in Xcode and checking the WebCore target in the details sidebar.  You can also manually edit this file to add something like this
```
E8FC921B285BFD5B004904B2 /* JSWindowEventHandlers+Gamepad.h in Headers */,
```

marcoscaceres replied:
Ok, I did it with the UI as you instructed just to be safe. Ut added it gave it a new hash and added it to headers, so hopefully that's right.

dcrousso commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -5531,6 +5531,8 @@
 		E7FBE7C224EB2FB000A18E62 /* StereoPannerNode.h in Headers */ = {isa = PBXBuildFile; fileRef = E7FBE7C124EB2FAB00A18E62 /* StereoPannerNode.h */; };
 		E7FBE7C324EB2FB600A18E62 /* StereoPannerOptions.h in Headers */ = {isa = PBXBuildFile; fileRef = E7FBE7C024EB2FAB00A18E62 /* StereoPannerOptions.h */; };
+		E8FC921B285BFD5B004904B2 /* JSWindowEventHandlers+Gamepad.h in Product Dependencies */ = {isa = PBXBuildFile; fileRef = E8FC9219285BFD5A004904B2 /* JSWindowEventHandlers+Gamepad.h */; };
+		E8FC921C285BFD5B004904B2 /* JSWindowEventHandlers+Gamepad.cpp in Product Dependencies */ = {isa = PBXBuildFile; fileRef = E8FC921A285BFD5B004904B2 /* JSWindowEventHandlers+Gamepad.cpp */; };
```
I dont think we want this in the `.xcodeproj` as it kinda negates us adding it to `Source/WebCore/Sources.txt`.  You should be able to uncheck the WebCore target in the details sidebar or manually edit this file to remove this line.

marcoscaceres committed:
Add ongamepadconnected + ongamepaddisconnected
https://bugs.webkit.org/show_bug.cgi?id=223860

Reviewed by Brady Eidson.

* LayoutTests/gamepad/gamepad-event-handlers-expected.txt: Added.
* LayoutTests/gamepad/gamepad-event-handlers.html: Added.
* LayoutTests/gamepad/gamepad-polling-access-expected.txt:
* LayoutTests/gamepad/gamepad-polling-access.html:
* LayoutTests/gamepad/gamepad-visibility-1.html:
* LayoutTests/platform/gtk/TestExpectations
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/gamepad/WindowEventHandlers+Gamepad.idl: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/HTMLAttributeNames.in:

Canonical link: https://commits.webkit.org/252331@main
